src/Build/Graph/ProjectInterpretation.cs

```diff
@@ -227,6 +230,10 @@ ProjectItemInstance projectReference
             // The properties on the project reference supersede the ones from the MSBuild task instead of appending.
             if (newProperties.Count == 0)
             {
+                if(projectReference.Project.GetPropertyValue(EnableDynamicPlatformResolutionMetadataName).Equals("true") && String.IsNullOrEmpty(projectReference.GetMetadataValue(SetPlatformMetadataName)))
```

Comment:
Use `ConversionUtilities.ConvertStringToBool` for `EnableDynamicPlatformResolutionMetadataName`

Reply:
Thank you for this tip. I implemented it